src/contig.rs

```diff
@@ -1,46 +1,63 @@
 use std::cmp::Ordering;
-use std::fmt::Debug;
-
-pub trait Contig: PartialOrd {
-    fn id(&self) -> usize;
-    fn name(&self) -> &str;
-    fn sequence_role(&self) -> &SequenceRole;
-    fn assigned_molecule(&self) -> &str;
-    fn assigned_molecule_type(&self) -> &AssignedMoleculeType;
-    fn length(&self) -> usize;
-    fn gen_bank_accession(&self) -> &str;
-    fn ref_seq_accession(&self) -> &str;
-    fn ucsc_name(&self) -> &str;
-}
+
+use crate::assigned_molecule_type::AssignedMoleculeType;
+use crate::sequence_role::SequenceRole;
+use crate::err::SvartError;
 
 #[derive(Debug)]
-pub struct UnknownContig {
+pub struct Contig {
     id: usize,
-    name: &'static str,
+    name: String,
     sequence_role: SequenceRole,
-    assigned_molecule: &'static str,
+    assigned_molecule: String,
     assigned_molecule_type: AssignedMoleculeType,
     length: usize,
-    gen_bank_accession: &'static str,
-    ref_seq_accession: &'static str,
-    ucsc_name: &'static str,
+    gen_bank_accession: String,
+    ref_seq_accession: String,
+    ucsc_name: String,
 }
+impl Contig {
+
+    pub fn of(id: usize,
+              name: String,
+              sequence_role: SequenceRole,
+              assigned_molecule: String,
+              assigned_molecule_type: AssignedMoleculeType,
+              length: usize,
+              gen_bank_accession: String,
+              ref_seq_accession: String,
+              ucsc_name: String) -> Result<Contig, SvartError> {
+        if id == 0 {
+            return Err(SvartError::IllegalValueError("Id 0 is reserved for the unknown contig"));
+        }
+
+        Ok(Contig {
+            id,
+            name,
+            sequence_role,
+            assigned_molecule,
+            assigned_molecule_type,
+            length,
+            gen_bank_accession,
+            ref_seq_accession,
+            ucsc_name,
+        })
+    }
 
-impl Contig for UnknownContig {
     fn id(&self) -> usize {
         self.id
     }
 
     fn name(&self) -> &str {
-        self.name
+        self.name.as_str()
```

Comment:
I think this can be also written as:

```rust
&self.name
```

However, I suspect that it will end up being the same thing anyway. So, no need to change anything here.